rclcpp/test/rclcpp/test_node.cpp

```diff
@@ -1444,6 +1444,38 @@ TEST_F(TestNode, set_parameter_undeclared_parameters_not_allowed) {
     EXPECT_EQ(value.get_type(), rclcpp::PARAMETER_STRING);
     EXPECT_EQ(value.get_value<std::string>(), "asd");
   }
+  {
+    // adding a parameter in "pre set parameter" callback, when that
+    // parameter has not been declared before will throw
+    auto name1 = "parameter"_unq;
+    auto name2 = "parameter"_unq;
+    auto default_value = 0;  // default value of name1 param
+
+    rcl_interfaces::msg::ParameterDescriptor descriptor;
+    descriptor.type = rclcpp::PARAMETER_INTEGER;
+
+    // declare name1 parameter only
+    node->declare_parameter(name1, default_value, descriptor);
```

Comment:
I think the descriptor is optional (the fact that it's an integer can be inferred by the default value).
```suggestion
    // declare name1 parameter only
    node->declare_parameter(name1, default_value);
```